packages/ui/src/elements/WhereBuilder/Condition/Relationship/index.tsx

```diff
@@ -76,18 +76,16 @@ export const RelationshipField: React.FC<Props> = (props) => {
         const fieldToSearch = collection?.admin?.useAsTitle || 'id'
         const pageIndex = nextPageByRelationshipRef.current.get(relationSlug)
 
-        const query: {
-          depth?: number
-          limit?: number
-          page?: number
-          where: Where
-        } = {
+        const query = {
           depth: 0,
           limit: maxResultsPerRequest,
           page: pageIndex,
+          select: {
+            [fieldToSearch]: true,
+          },
           where: {
             and: [],
-          },
+          } as Where,
```

Comment:
- simplified types using inference when possible
- use select to bring only the field we are interested in

Reply:
Do we need to use a type inference here? Can we just type the object outright?

Reply:
It is preferable to use inference because
1. it is fewer lines of code and
2. it is more type-safe. Explicit types can narrow the type more than necessary.

Reply:
> it is more type-safe. 

I think the opposite is true. With assertions, the compiler essentially _trusts_ that the object is of the type you say it is. Even if there are required properties, the compiler skips validation. It should be explicitly typed like this instead:

```ts
const query: Where = {
  // ...
}
```

Reply:
Yes, that's true, in the case of Where it's a typecast not inference. We should have a Query type somewhere.
I only found JoinQuery which doesn't seem appropriate...